### PR TITLE
Liballoc extend use intra doc link

### PR DESCRIPTION
**Pull request by pickfire**
Superseeds https://github.com/rust-lang/rust/pull/75729/commits

r? @jyn514

### CODE REVIEW

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -1610,8 +1610,6 @@ impl<T: Clone> Vec<T> {
     /// vec.extend_from_slice(&[2, 3, 4]);
     /// assert_eq!(vec, [1, 2, 3, 4]);
     /// ```
```
Comment (jyn514) [inline library/alloc/src/vec.rs:1612]:
```suggestion
    /// ```
    ///
    /// [`extend`]: Vec::extend
```

**Review by jyn514 — COMMENTED** [review]
r=me once the link is fixed. Currently it will give an error.

File: library/alloc/src/vec.rs (hunk 1 above)
Comment (pickfire) [reply library/alloc/src/vec.rs:1612]: Wait, that's the only thing this patch does. I thought you say it can be removed already?

Comment (jyn514) [reply library/alloc/src/vec.rs:1612]: No, there is an issue open to _add_ this feature (https://github.com/rust-lang/rust/issues/59039), but it is not yet implemented. My suggested change goes from `#method.extend` to the associated function.

**Pull request merged**